api/v08/kogitoserverlessplatform_types.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v08
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ConfigurationSpec represents a generic configuration specification
+type ConfigurationSpec struct {
+	// represents the type of configuration, ie: property, configmap, secret, ...
+	Type string `json:"type"`
+	// the value to assign to the configuration (syntax may vary depending on the `Type`)
+	Value string `json:"value"`
+}
+
+// PlatformCluster is the kind of orchestration cluster the platform is installed into
+type PlatformCluster string
+
+// BuildStrategy specifies how the Build should be executed.
+// It will trigger a Kaniko pod that will take care of producing the expected image containing the Quarkus/Kogito runtime.
+// +kubebuilder:validation:Enum=pod
+type BuildStrategy string
+
+const (
+	// BuildStrategyPod performs the build in a `Pod` (will schedule a new builder ephemeral `Pod` which will take care of the build action).
+	BuildStrategyPod BuildStrategy = "pod"
+)
+
+// BuildStrategies is a list of strategies allowed for the build
+var BuildStrategies = []BuildStrategy{
+	BuildStrategyPod,
+}
+
+type BuildPublishStrategy string
+
+const (
+	// BuildPublishStrategyKaniko uses Kaniko project (https://github.com/GoogleContainerTools/kaniko)
+	// in order to push the incremental images to the image repository. It can be used with `pod` BuildStrategy.
+	BuildPublishStrategyKaniko BuildPublishStrategy = "Kaniko"
+)
+
+// BuildPublishStrategies the list of all available publish strategies
+var BuildPublishStrategies = []BuildPublishStrategy{
+	BuildPublishStrategyKaniko,
+}
+
+// RegistrySpec provides the configuration for the container registry
+type RegistrySpec struct {
```

Comment:
I think we need to add the kubebuilder annotations on all objects that should be listed in the crd.